### PR TITLE
Olympia merge from master

### PR DESCRIPTION
**Pull request by mnaamani**
Update olympia from master.

- Brings in new github workflows and improved aws cloud formation scripts.
- Latest patches to types and metadata protobuf to build target es2017.
- Update runtime code shasum script to use `gnu-sed` on OSX to fix error when using version of `sed` that comes by default on OSX:

```
TEST_NODE=1 ./scripts/runtime-code-shasum.sh 
sed: 1: "$a1000": command a expects \ followed by text
da39a3ee5e6b4b0d3255bfef95601890afd80709
```

Note: The workflows are ported as is, but some need customization for olympia and most certainly the ansible playbooks need to be updated to work correct with new olympia node and chainspec builder.

cc: @ahhda

### CODE REVIEW

File: start.sh [hunk 1]
```diff
@@ -28,9 +28,9 @@ docker-compose up -d joystream-node
 export SKIP_MOCK_CONTENT=true
 export SKIP_QUERY_NODE_CHECKS=true
 HOST_IP=$(tests/network-tests/get-host-ip.sh)
-export COLOSSUS_1_URL="http://${HOST_IP}:3333"
+export COLOSSUS_1_URL=${COLOSSUS_1_URL:="http://${HOST_IP}:3333"}
 export COLOSSUS_1_TRANSACTOR_KEY=$(docker run --rm --pull=always docker.io/parity/subkey:2.0.1 inspect ${COLOSSUS_1_TRANSACTOR_URI} --output-type json | jq .ss58Address -r)
-export DISTRIBUTOR_1_URL="http://${HOST_IP}:3334"
+export DISTRIBUTOR_1_URL={DISTRIBUTOR_1_URL:="http://${HOST_IP}:3334"}
```
Comment (mnaamani) [inline start.sh]:
```suggestion
export DISTRIBUTOR_1_URL=${DISTRIBUTOR_1_URL:="http://${HOST_IP}:3334"}
```

**Comment by mnaamani** [comment]
I pushed the same work into olympia-playground branch to test out all the github actions:
https://github.com/Joystream/joystream/actions?query=branch%3Aolympia-playground

First priority will be to get build-node-docker workflow working after this merge before going deeper into the other fixes needed.

**Review by ahhda — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -77,7 +77,7 @@ A step by step guide to setup a full node and validator on the Joystream testnet
 ### Integration tests
 
 ```bash
-tests/network-tests/run-tests.sh
+tests/integration-tests/run-full-tests.sh
```
Comment (Lezek123) [inline README.md:80]: This script is not yet updated, I was wondering if we should move back to `network-tests` [once the merge process of the integration tests](https://github.com/Joystream/joystream/issues/3015) is done?

**Review by Lezek123 — APPROVED** [review]
LGTM, just left a small question

File: README.md (hunk 2 above)
Comment (mnaamani) [reply README.md:80]: Yeah I wasn't sure which one we wanted to use keep. We can certainly update this once the merge is complete.

**Pull request merged**